SmartDeviceLink/SDLLockScreenPresenter.m

```diff
@@ -31,241 +31,173 @@ - (instancetype)init {
-        }
-
-        NSArray<UIWindow *> *windows = appWindowScene.windows;
-        UIWindow *appWindow = nil;
-        for (UIWindow *window in windows) {
-            SDLLogV(@"Checking window: %@", window);
-            if (window != self.lockWindow) {
-                appWindow = window;
-                break;
-            }
-        }
-
-        [self sdl_dismissWithAppWindow:appWindow];
-    }
-}
-
-- (void)sdl_dismissWithAppWindow:(nullable UIWindow *)appWindow {
-    if (appWindow == nil) {
-        SDLLogE(@"Unable to find the app's window");
-        return;
-    } else if (appWindow.isKeyWindow) {
-        SDLLogW(@"Attempted to dismiss lock screen, but it is already dismissed");
-        return;
-    } else if (self.lockViewController == nil) {
-        SDLLogW(@"Attempted to dismiss lock screen, but lockViewController is not set");
-        return;
+    if ([self sdl_isDismissing]) {
+        // Make sure we are not already animating, otherwise the app may crash
+        SDLLogV(@"The lockscreen is already being dismissed");
+        if (completionHandler == nil) { return; }
+        return completionHandler();
     }
 
-    // Let us know we are about to dismiss.
+    // Let ourselves know that the lockscreen will dismiss so we can pause video streaming for a few milliseconds - otherwise the animation to dismiss the lockscreen will be very janky.
     [[NSNotificationCenter defaultCenter] postNotificationName:SDLLockScreenManagerWillDismissLockScreenViewController object:nil];
 
-    // Dismiss the lockscreen
-    SDLLogD(@"Dismiss lock screen window from app window: %@", appWindow);
+    SDLLogD(@"Hiding the lockscreen window");
+    __weak typeof(self) weakSelf = self;
     [self.lockViewController dismissViewControllerAnimated:YES completion:^{
-        CGRect lockFrame = self.lockWindow.frame;
-        lockFrame.origin.x = CGRectGetWidth(lockFrame);
-        self.lockWindow.frame = lockFrame;
-
-        // Quickly move the map back, and make it the key window.
-        appWindow.frame = self.lockWindow.bounds;
-        [appWindow makeKeyAndVisible];
+        [weakSelf.lockWindow setHidden:YES];
 
-        // Tell ourselves we are done.
+        // Tell everyone we are done so video streaming can resume
         [[NSNotificationCenter defaultCenter] postNotificationName:SDLLockScreenManagerDidDismissLockScreenViewController object:nil];
+
+        if (completionHandler == nil) { return; }
+        return completionHandler();
     }];
 }
 
 
-#pragma mark - isPresented Getter
+#pragma mark - Custom Presented / Dismissed Getters
 
-- (BOOL)presented {
-    __block BOOL isPresented = NO;
-    if ([NSThread isMainThread]) {
-        isPresented = [self sdl_presented];
-    } else {
-        dispatch_sync(dispatch_get_main_queue(), ^{
-            isPresented = [self sdl_presented];
-        });
-    }
+/// Returns whether or not the lockViewController is presented or currently animating
+- (BOOL)sdl_isPresentedOrPresenting {
```

Comment:
Instead of private functions, this could be a private property with this as the getter. Then you can get rid of the `sdl_` and refer to it with dot-syntax.

Reply:
Fixed.